### DISCUSSION TITLE
brand new UI with Angular

### DISCUSSION THREAD

**Discussion by pishangujeniya** [d3vilh/openvpn-ui#24] [Ideas]
I am planning to develop the UI From scratch, if you provide API as a service from your GoLang backend then it would be better.

I am going to start development with https://github.com/akveo/ngx-admin  - [demo](https://www.akveo.com/ngx-admin/pages/dashboard?theme=default)

What are your thoughts on this?

**Comment by d3vilh** [comment]
I find Ngx-admin quite appealing.
OpenVPN UI, relies on [AdminLTE2](https://adminlte.io/themes/AdminLTE/index2.html) framework as its foundation.

There are plans in the pipeline to upgrade it to [AdminLTE3](https://adminlte.io/), though.

As API I make use of the standard OpenVPN management console, which is essentially the first line in the server configuration file [server.conf](https://github.com/d3vilh/openvpn-server/blob/main/config/server.conf#L1). For managing the Certificate Authority (CA) and certificates, I employ EasyRSA and trigger these actions with a few bash scripts.

Currently, OpenVPN UI has all the necessary JavaScript and CSS files in the [`static`](https://github.com/d3vilh/openvpn-ui/tree/main/static) directory, and the [`Login.html`](https://github.com/d3vilh/openvpn-ui/blob/main/views/login.html#L17) page has already been updated to version 3. However, the rest of the pages still use version 2, resulting in a somewhat inconsistent appearance due to the combination of new JavaScript and old CSS. This migration project will require some additional time, which I'm currently short on.

If you're up for a challenge, you can help upgrade the OpenVPN UI to use [AdminLTE version 3](https://adminlte.io/docs/3.2/upgrade-guide.html). I would greatly appreciate any pull requests you can contribute.

**Reply by pishangujeniya** [reply]
@d3vilh 

Absolutely! We can split our responsibilities, with one of us concentrating on backend API development and the other on shaping the Angular UI. This not only eliminates tight coupling but also ensures dedicated focus on each aspect of the project.

Furthermore, since your repository and project are still in their early stages, it's essential that we adhere to established open-source community practices. By doing so, once our project is complete, it will seamlessly integrate into the broader ecosystem. This will facilitate collaboration, contributions from others, and the extension of functionality to create customized mobile and web apps. The potential for our project's future is truly exciting!

> I am thinking of developing a containerized scalable backend using the latest .NET ecosystem.
